src/control/server/storage/scm/provider_test.go

```diff
@@ -817,10 +974,18 @@ func TestProviderFormat(t *testing.T) {
 					}
 					t.Fatalf("%s leaked from IsMounted() check", err)
 				default:
-					if tc.expErr != nil && tc.expErr.Error() == errors.Cause(err).Error() {
-						return
+					if tc.expErr == nil {
+						t.Fatal(err)
 					}
-					t.Fatal(err)
+
+					// expErr will contain the original mountpoint
+					errStr := errors.Cause(err).Error()
+					errStr = strings.Replace(errStr, testDir, "", 1)
```

Comment:
IMO this is a little counterintuitive. Wouldn't it be better to update expErr (maybe use a helper function to prepend the testdir in the tc definition?) than to modify the error we got back?

Reply:
whilst this sounds simple I spent a while trying to fix but ended up something just as nasty, going to leave as is if that's okay, it provides the required coverage even if it doesn't look nice

Reply:
That's reasonable. Thanks.